src/lib.rs

```diff
@@ -601,7 +601,27 @@ impl<'a> WrapIterImpl<'a> {
         while let Some((idx, ch)) = self.char_indices.next() {
             let char_width = ch.width().unwrap_or(0);
             let char_len = ch.len_utf8();
-            if is_whitespace(ch) {
+
+            if ch == '\n' {
+                self.split = idx;
+                self.split_len = char_len;
+                self.line_width_at_split = self.line_width;
+                self.in_whitespace = false;
+
+                // If this is not the final line, return the current line. Otherwise,
+                // we will return the line with its line break after exiting the loop
+                if self.split + self.split_len < self.source.len() {
```

Comment:
I didn't actually try, but would it be possible to combine the logic here with the existing logic that kicks in when the current line it too long? Maybe by changing the conditions slightly:
```
if ch != '\n' && is_whitespace(ch) {
    // ...
} else if ch == `\n` || self.line_width + char_width > wrapper.width {
    // ...
}
```

No idea if it works! :-D

Reply:
hmm maybe but I can't really seem to get it working this way ^^ feel free to commit if you find a way